### PR TITLE
Don't mark index tasks as deleted when segment is compacted

### PR DESCRIPTION
**Pull request by xiaocai2333**
Signed-off-by: cai.zhang <cai.zhang@zilliz.com>
issue: #18642 , #19164, #19143

### CODE REVIEW

**Comment by xiaocai2333** [comment]
rerun ut

**Comment by xiaocai2333** [comment]
rerun ut

File: internal/indexcoord/flush_segment_watcher.go [hunk 1]
```diff
@@ -311,76 +213,56 @@ func (fsw *flushedSegmentWatcher) setInternalTaskSegmentInfo(segID UniqueID, seg
...
-		return
-	}
 	t := fsw.getInternalTask(segID)
 	log.Debug("IndexCoord flushedSegmentWatcher process internal task", zap.Int64("segID", segID),
 		zap.String("state", t.state.String()))
 
 	switch t.state {
+	case indexTaskPrepare:
+		if err := fsw.prepare(segID); err != nil {
+			log.Error("flushedSegmentWatcher prepare internal task fail", zap.Int64("segID", segID), zap.Error(err))
```
Comment (czs007) [inline internal/indexcoord/flush_segment_watcher.go:240]: log.Warn

File: internal/indexcoord/flush_segment_watcher.go [hunk 2]
```diff
@@ -311,76 +213,56 @@ func (fsw *flushedSegmentWatcher) setInternalTaskSegmentInfo(segID UniqueID, seg
...
+			return
+		}
+		fsw.updateInternalTaskState(segID, indexTaskInit)
 	case indexTaskInit:
-		fsw.constructTask(t)
+		if err := fsw.constructTask(t); err != nil {
+			log.Error("flushedSegmentWatcher construct task fail", zap.Int64("segID", segID), zap.Error(err))
```
Comment (czs007) [inline internal/indexcoord/flush_segment_watcher.go:246]: log.Warn

File: internal/indexcoord/flush_segment_watcher.go [hunk 3]
```diff
@@ -587,30 +411,40 @@ func (fsw *flushedSegmentWatcher) pullSegmentInfo(segmentID UniqueID) error {
...
-	for _, segID := range t.segmentInfo.CompactionFrom {
-		fsw.deleteChildrenTask(segID)
-		if _, ok := fsw.internalTasks[segID]; ok {
-			fsw.updateInternalTaskState(segID, indexTaskDeleted)
-		}
-	}
-}
+//func (fsw *flushedSegmentWatcher) removeCompactedTasks(t *internalTask) error {
+//	log.Debug("IndexCoord flushedSegmentWatcher mark task as deleted which is compacted", zap.Int64("segID", t.segmentInfo.ID),
+//		zap.Int64s("compactionFrom", t.segmentInfo.CompactionFrom))
```
Comment (czs007) [inline internal/indexcoord/flush_segment_watcher.go:437]: just remove this.

**Pull request merged**